File: 0119_pascals_triangle_ii.py
# https://leetcode-cn.com/problems/pascals-triangle-ii/

class Solution:
    def getRow(self, rowIndex):
        """
        Inputs:
            rowIndex: int
        Outputs:
            res: list[int]
        """
        
        res = [1]
        for i in range(1,rowIndex+1):
            res.insert(0,0)
            for j in range(i):
                res[j] = res[j] + res[j+1]
        return res

File: Cpp/0119_pascals_triangle_ii.cpp
// https://leetcode.cn/problems/pascals-triangle-ii/

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> res(rowIndex+1, 1);
        for (int row = 2; row <= rowIndex; ++row) {
            int last = 1;
            for (int i = 1; i < row; ++i) {
                int tmp = last + res[i];
                last = res[i];
                res[i] = tmp;
            }
        }
        return res;
    }
};

int main() {
    Solution solu;
    int rowIndex = 5;
    vector<int> res = solu.getRow(rowIndex);
    for (int i: res) {
        cout << i << ", ";
    }
    cout << '\n';
    
    return 0;
}